module2/projects/coupon-codes-api/index.md

```diff
@@ -0,0 +1,467 @@
+
+## Details
+
+* This is a solo project, to be completed alone without assistance from cohortmates, alumni, mentors, rocks, etc.
+* Must use Rails 7.1.x and Ruby 3.2.2. 
+* Additional gems to be added to the project must have instructor approval.
+  * RSpec, Pry, Simplecov, Shoulda-Matchers, Orderly, jsonapi-serializer (or other serializer gem), HTTParty, Faker, Faraday, and FactoryBot are pre-approved
+* Scaffolding is not permitted on this project.
+
+## Setup
+
+This project is an extension of Little Shop. Students have two options for setup:
+
+1. If your Little Shop project is complete, you can use it as a starting point for this project. If you are not the repo owner, fork the project to your account. If you are the repo owner, you can work off the repo without forking, just make sure your teammates have a chance to fork before pushing any commits to your repo.
+1. If your Little Shop project is _not_ complete, fork **[this repo](https://github.com/turingschool-examples/little-shop-be)** as a starting point for this project.
+
+## Evaluation
+Evaluation information for this project can be found [here](./evaluation).
+
+-----
+
+## Functionality Overview
+
+* A Coupon belongs to a Merchant
+* An Invoice optionally belongs to a Coupon. An invoice may only have one coupon.
+  * Note: When creating this new association on Invoice, your existing tests will fail unless the association is optional. Use these guides as a reference.
+  * You are not required to build functionality for a user applying a coupon to an invoice, but can instead use test data, Rails console or seed data to add coupons to existing invoices to verify behavior.
+* You should build full CRUD functionality for coupons with criteria/restrictions defined below:
+   - A merchant can have a maximum of 5 activated coupons in the system at one time.
+   - A merchant cannot delete a coupon, rather they can activate/deactivate them.
+   - A Coupon has a name, unique code (e.g. "BOGO50"), and either percent-off or dollar-off value. The coupon's code must be unique in the whole database.
+* If a coupon's dollar value (ex. "$10 off") exceeds the total cost of that merchant's items on the invoice, the grand total for that merchant's items should then be $0. (In other words, the merchant will never *owe* money to a customer.)
+* A coupon code from a Merchant only applies to Items sold by that Merchant.
+
+​
+## 1) CRUD Endpoints
+
+Below is the expected JSON response for each request. We have also outlined a few examples of Sad Paths you may consider adding in. In your project, you should take time to implement at least 2 sad paths total, but you are not limited to the examples we provide. 
+
+<section class="dropdown">
+### 1. Merchant Coupon Show Page 
+
+Returns a specific coupon and shows a count of how many times that coupon has been used.
+
+**Request**
+```bash
+GET /api/v1/merchants/:merchant_id/coupons/:id
```

Comment:
To @abdulredd's comment, I guess we could just have the endpoint description (without the `GET /api/v1/merchants/:merchant_id/coupons/:id` explicitly listed for them) and they have to determine the most RESTful path? I could go either way on this

Reply:
I'll go ahead an remove the request examples for the CRUD endpoints.